source/common/access_log/access_log_formatter.cc

```diff
@@ -132,10 +155,19 @@ std::vector<FormatterPtr> AccessLogFormatParser::parse(const std::string& format
         absl::optional<size_t> max_length;
         const size_t start = 5;
 
-        parseCommand(token, start, main_header, alternative_header, max_length);
+        parseCommandHeader(token, start, main_header, alternative_header, max_length);
 
         formatters.emplace_back(
             FormatterPtr(new ResponseHeaderFormatter(main_header, alternative_header, max_length)));
+      } else if (token.find(DYNAMIC_META_TOKEN) == 0) {
+        std::string filter_namespace;
+        absl::optional<size_t> max_length;
+        std::vector<std::string> path;
+        const size_t start = DYNAMIC_META_TOKEN.size();
+
+        parseCommand(token, start, ":", filter_namespace, path, max_length);
+        formatters.emplace_back(
+            FormatterPtr(new DynamicMetadataFormatter(filter_namespace, path, max_length)));
```

Comment:
You can probably skip the explicit `FormatterPtr` here, since this is `emplace_back`.

Reply:
Will take a look. Note that the existing formatters also use this pattern.